README.md

```diff
@@ -0,0 +1,123 @@
+meta-rust-bin
+=============
+
+An OpenEmebdded/Yocto layer providing pre-built toolchains for the
+[Rust programming language](https://www.rust-lang.org).
+
+<!-- toc -->
+
+- [Basic Example](#basic-example)
+- [Features](#features)
+- [Advanced Features](#advanced-features)
+  * [Specifying Cargo Features](#specifying-cargo-features)
+  * [Using Components Individually](#using-components-individually)
+- [Pre-built vs. Compiled](#pre-built-vs-compiled)
+- [Copyright](#copyright)
+
+<!-- tocstop -->
+
+
+## Basic Example
+
+A basic class for cargo-based executables is provided. The following is a
+simple recipe that builds the [gpio-utils](https://github.com/rust-embedded/gpio-utils)
+crate from a branch tagged with the version `${PV}`:
+
+```bitbake
+inherit cargo
+
+SUMMARY = "GPIO Utilities"
+HOMEPAGE = "git://github.com/rust-embedded/gpio-utils"
+LICENSE = "MIT"
+
+SRC_URI = "https://github.com/rust-embedded/gpio-utils.git;tag=${PV}"
+S = "${WORKDIR}/git"
+
+LIC_FILES_CHKSUM = "file://LICENSE-MIT;md5=935a9b2a57ae70704d8125b9c0e39059"
+```
+
+As you can see, there is almost no overhead introduced from the `cargo` class
+beyond simply inheriting it. The `cargo` class adds the appropriate Rust
+dependencies as well as default compile and install steps.
+
+
+## Features
+
+Currently supported:
+
+  * Rust 1.11
+  * x86 (32 and 64-bit), ARM (32 and 64-bit) build systems.
+  * All Linux architectures that Rust itself supports (Multiple flavors of:
+    x86, ARM, PPC, and MIPS)
+
```

Comment:
No shared libraries are installed by default, including libstd, since all Rust dependencies are statically linked by default in Cargo. There's an option, `-C prefer-dynamic`, that could be added to the Cargo config file and would allow a dynamically linked executable. That could definitely be useful, so I'm adding dynamic linking as a future enhancement. I also clarified the README to make it explicit what the current linking state is: 2d5e033